### ISSUE TITLE
YT code is loaded even if there is no video

### ISSUE THREAD

**Issue by morvy** [duracelltomi/gtm4wp#246]
Hi,

I noticed that after turning on YT tracking/experimental, plugin is loading scripts from youtube.com on every page.
For Gutenberg embed, it should be enough to check for video content with `has_block`, but I'm not sure about Classic Editor and iframes. Regular expression would be the last resort I think, but on cached content it doesn't really matter that much.

**Comment by duracelltomi** [comment]
Script is loaded from here:
https://github.com/duracelltomi/gtm4wp/blob/master/js/gtm4wp-youtube.js

What I could imagine is to add a dom ready listener and then look at YT iframes in the DOM and only load this code if there are any YT embeds.

**Comment by duracelltomi** [comment]
I am not sure the performance of this code will be good. It scans through the whole post content on the server side increasing processing time and thus, response time.

Checking this in the browser seems to be a better solution for me.

**Comment by morvy** [comment]
Yes this will definitely put some load on the server, but I believe every WP site now uses some kind of caching - if not, PSI is looking bad and those numbers matter to managers :) So if it's checked on server and then saved in cache in pure html (W3TC, Super Cache, WP Rocket..) it shouldn't add load on server anymore.
I don't like additional scripts because JS is still single threaded and it can block execution of other JS. I'm pretty bad in JS, so I just avoid using JS whenever possible, because when I use it, it has impact on performance - mostly on mobiles.

**Comment by duracelltomi** [comment]
Here you are: https://github.com/duracelltomi/gtm4wp/commit/f3fa21059de6eef6b9d63a364e42711c46d8136c
